lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -1,38 +1,78 @@
 import ErrorDialog from './ErrorDialog';
 import Spinner from './Spinner';
 import SvgIcon from './SvgIcon';
+import { fetchGrade, submitGrade } from '../utils/grader-service';
 import { trustMarkup } from '../utils/trusted';
+import { formatToNumber, validateGrade } from '../utils/validation';
 import ValidationMessage from './ValidationMessage';
 
+// Grades are always stored as a value between [0-1] on the service layer.
+// Scale the grade to [0-10] when loading it into the UI and scale it
+// back to [0-1] when saving it to the api service.
+const GRADE_MULTIPLIER = 10;
+
+/**
+ * Custom useEffect function that handles fetching a student's
+ * grade and returning the result of that grade and the loading
+ * and error status of that fetch request.
+ */
+const useFetchGrade = student => {
+  const { authToken } = useContext(Config);
+  const [grade, setGrade] = useState('');
+  const [gradeLoading, setGradeLoading] = useState(false);
+
+  useEffect(() => {
+    if (Object.entries(student).length) {
+      // Don't bother fetching if the student object is empty
+      // Fetch the grade from the service api
+      // See https://www.robinwieruch.de/react-hooks-fetch-data for async in useEffect
+      const fetchData = async () => {
+        setGradeLoading(true);
+        setGrade(''); // Clear previous grade so we don't show the wrong grade with the new student
+        const response = await fetchGrade({ student, authToken });
+        setGrade(response.currentScore * GRADE_MULTIPLIER);
+        setGradeLoading(false);
+      };
+      fetchData();
+    }
+  }, [student, authToken]);
+  return [{ grade, gradeLoading }];
+};
+
 /**
  * A form with a single input field and submit button for an instructor to
  * save a students grade.
  */
 
 export default function SubmitGradeForm({ disabled = false, student }) {
-  const [showError, setShowError] = useState(false); // Is there a validation error message to show?
-  const [gradeErrorMessage, setGradeErrorMessage] = useState(''); // The actual validation error message.
+  // State for loading the grade
+  const [{ grade, gradeLoading }] = useFetchGrade(student);
+
+  // State for saving the grade
+  const [submitGradeError, setSubmitGradeError] = useState(''); // If there is an error when submitting a grade
+  const [gradeSaving, setGradeSaving] = useState(false); // True when the grade is being currently being posted to the service
+  const [gradeSaved, setGradeSaved] = useState(false); // Changes the input field background to green for a short duration when true
 
-  const [networkError, setNetworkError] = useState(''); // if there is an error when submitting grade
-  const [requestStatus, setRequestStatus] = useState(''); // ajax request state, one of ('', 'fetching', 'error', 'done')
+  // State for local validation errors
+  const [showValidationError, setValidationError] = useState(false); // Is there a validation error message to show?
```

Comment:
I need 2 values here because we need to keep the validation text around while the message is closing (animation) so the text has to be independent of the closed state.